### PR TITLE
Implementing #63

### PR DESCRIPTION
**Pull request by frcroth**
Implementing users story #63. Added an info button. Might want to combine Inspect Document button with this new button.

### CODE REVIEW

**Comment by frcroth** [comment]
Remove magic numbers

File: packages/RichTextEditing-Core.package/RichTextDocument.class/instance/charcountExcept..st [hunk 1]
```diff
@@ -0,0 +1,14 @@
+count
+charcountExcept: aListOfBytecodes
+	
+	| byteString count i|
+	count := 0.
+	byteString := self text string asByteArray.
+	
+	i := 1.
+	[i <= byteString size] whileTrue:[
+		(aListOfBytecodes includes: (byteString at: i)) ifFalse: [count := count+1].
```
Comment (oleschl) [inline packages/RichTextEditing-Core.package/RichTextDocument.class/instance/charcountExcept..st]: ^ byteString count: [:each | (aListOfBytecodes includes: each) not]

File: packages/RichTextEditing-Core.package/RichTextDocument.class/instance/emptyChars.st [hunk 2]
```diff
@@ -0,0 +1,4 @@
+count
+emptyChars
+
+	^ #(182 32 10 13)
```
Comment (oleschl) [inline packages/RichTextEditing-Core.package/RichTextDocument.class/instance/emptyChars.st]: what about a method:
``` 
spaceValue 
^32 
```
as in RichTextEditorModel?

Comment (frcroth) [reply packages/RichTextEditing-Core.package/RichTextDocument.class/instance/emptyChars.st]: what do you mean? do you want the method to be more specific?

File: packages/RichTextEditing-Core.package/RichTextDocument.class/instance/charcountExcept..st (hunk 1 above)
Comment (frcroth) [reply packages/RichTextEditing-Core.package/RichTextDocument.class/instance/charcountExcept..st]: nice!, will be changed

File: packages/RichTextEditing-Core.package/RichTextDocument.class/instance/wordcount.st [hunk 3]
```diff
@@ -1,16 +1,15 @@
...
-	splitChars := #(182 32 10 13).
 	
 	i := 1.
 	[i <= byteString size] whileTrue:[
-		(readWord and: (splitChars includes: (byteString at: i))) ifTrue: [readWord := false. count := count+1].
-		(splitChars includes: (byteString at: i)) ifFalse: [readWord := true].
+		(readWord and: (self emptyChars includes: (byteString at: i))) ifTrue: [readWord := false. count := count+1].
+		(self emptyChars includes: (byteString at: i)) ifFalse: [readWord := true].
 		i := i + 1
 		].
```
Comment (oleschl) [inline packages/RichTextEditing-Core.package/RichTextDocument.class/instance/wordcount.st:14]: just tested this small code snippet which checks if the previous element is an emptyChar and the current not:
```
| prev bool |
prev := 32.
^ (#[104 97 108 32 108 32 32 111] count: [:each | bool := ((#[32] includes: each) not and: (#[32] includes: prev)). prev := each. bool ]) 
```
Not sure if better, however more smalltalk like

File: packages/RichTextEditing-Core.package/RichTextDocument.class/instance/wordcount.st [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+	readWord := false.
+	byteString := self text string asByteArray.
+	
+	i := 1.
+	[i <= byteString size] whileTrue:[
+		(readWord and: (self emptyChars includes: (byteString at: i))) ifTrue: [readWord := false. count := count+1].
+		(self emptyChars includes: (byteString at: i)) ifFalse: [readWord := true].
+		i := i + 1
+		].
+	readWord ifTrue: [count := count +1].
```
Comment (oleschl) [inline packages/RichTextEditing-Core.package/RichTextDocument.class/instance/wordcount.st]: just tested this small code snippet which checks if the previous element is an emptyChar and the current not:
```
| prev bool |
prev := 32.
^ (#[104 97 108 32 108 32 32 111] count: [:each | bool := ((#[32] includes: each) not and: (#[32] includes: prev)). prev := each. bool ]) 
```
Not sure if better, however more smalltalk like

Comment (frcroth) [reply packages/RichTextEditing-Core.package/RichTextDocument.class/instance/wordcount.st]: May not detect words appropriately when words are seperated by multiple seperator chars

Comment (oleschl) [reply packages/RichTextEditing-Core.package/RichTextDocument.class/instance/wordcount.st]: If both previous and current are an empty charstring the bool returns false and is not counted. I think every word is an empty charstring followed by a not empty charstring.

File: packages/RichTextEditing-Core.package/RichTextDocument.class/instance/emptyChars.st (hunk 2 above)
Comment (sirkrypt0) [reply packages/RichTextEditing-Core.package/RichTextDocument.class/instance/emptyChars.st]: Yea, or maybe a comment stating what each char is

**Review by sirkrypt0 — CHANGES_REQUESTED** [review]

File: packages/RichTextEditing-Core.package/RichTextDocument.class/instance/wordcount.st (hunk 4 above)
Comment (frcroth) [reply packages/RichTextEditing-Core.package/RichTextDocument.class/instance/wordcount.st]: implemented with 
350982f

File: packages/RichTextEditing-Core.package/RichTextDocument.class/instance/emptyChars.st (hunk 2 above)
Comment (frcroth) [reply packages/RichTextEditing-Core.package/RichTextDocument.class/instance/emptyChars.st]: > 
> 
> Yea, or maybe a comment stating what each char is

implemented in 
350982f

**Review by sirkrypt0 — APPROVED** [review]

**Pull request merged**